### DISCUSSION TITLE
BlockId incoherent

### DISCUSSION THREAD

**Discussion by d4rk0snet** [capsule-corp-ternoa/ternoa-hub#25] [Q&A] [answered]
Hello Everyone,

i noticed something strange in blockId sequence. 
The first block according to dictionnary was minted on timestamp 1649801556 (13/04/2022, 00:12:36), i confirmed that with that query on dictionnary : 
```
{
          extrinsics(
            first: 1,
            orderBy: [BLOCK_HEIGHT_DESC]
            filter: {
              and: [
                { 
                  blockId: {equalTo: "1"}
                  #signer: { equalTo: "5HbRivzefdVrxsRqyYXZSUGzHBsW53odx1FFtUxykyku5FjH"}
                  #timestamp: {equalTo: "2022-09-15T22:17:30"}        
                }
               ]
            }
          )
          {nodes { argsValue signer block {timestamp}}}
}
```

Now, with another query on events below :

```
{
          events(
            first: 1
            orderBy: [BLOCK_HEIGHT_DESC] 
            filter: {
              and: [
                {
                  event: {equalTo: "Rewarded"},
                  argsValue: {contains: "5GYnXQ3kwQuZeaGJiBt41AnJRokGyZmwjCTAczXUWjBQDRe4"},
                  
                  #timestamp: {equalTo: "2022-09-15T22:17:30"}
                }
              ]
            }
          )
          {nodes {blockId argsName argsValue block{timestamp number}}}
}
```

I got this :

```
{
  "data": {
    "events": {
      "nodes": [
        {
          "blockId": "2245092",
          "argsName": [
            "AccountId32",
            "u128"
          ],
          "argsValue": [
            "5GYnXQ3kwQuZeaGJiBt41AnJRokGyZmwjCTAczXUWjBQDRe4",
            "3556487677984837"
          ],
          "block": {
            "timestamp": "2022-09-15T22:17:30",
            "number": 2245092
          }
        }
      ]
    }
  }
}
```

So the block number which is equal to the blockId is : 2245092 and the timestamp 2022-09-15T22:17:30.
If i compute the number of seconds elapsed between 2022-09-15T22:17:30 - 2022-04-13T00:12:36 , i get : 13 471 494 (i used this website : http://www.my2lbox.com).

So with a block generation every 6s the block id should be : 13471494/6 = **2245249**

Why is there a difference between ids ?

**Comment by markopoloparadox** [comment, marked as answer]
Hello @d4rk0snet 
That's a very interesting question!

There is just a small flaw in your hypothesis. You assume that the block time is 6 seconds which isn't technically true.
You see, the chain in general tries to produce a block every 6 seconds but that's not absolute. The goal is to make sure that the average time is 6 seconds so this means that it's perfectly normal to have block times which are longer or shorter than 6 seconds. 
Example: When a runtime upgrade happens it usually take a minute for a new block to be produced. 

This means that in general you cannot say that a specific block in the future will happen on a specific time and instead approximations are used.

I just want to outline the fact that this issue is common across all blockchains. if you take a look at Solana, since 2020 they had 7 network outages, from which the longest outage halted the chain for 17 hours.

**Reply by d4rk0snet** [reply]
I see ! 
Everythings clear then. 
That's true that i was hoping to match timestamp with blockId but i will find another way.

Merci Marko :)
